src/annotator-content-analysis.js

```diff
@@ -1,54 +1,136 @@
-function annotatorCustomEditor(options){
+function annotatorContentAnalysis(options){
   return {
     start: function() {
       this.options = options;
-      this.getDataPayload();
+      this._elements();
+      this.getTemplates();
+      this.getData();
     },
 
-    // ajax get to retrieve data
-    getDataPayload: function(){
-      $.get(this.options.endpoint, (data) => {
-        this.configureData(data);
-      })
+    _elements: function() {
+      this.survey = {
+        answers: [],
+        topics: [],
+        questions: []
+      }
+    },
+
+    getTemplates: function() {
+      this.templates = {}
+      $.each(this.options.templates, (key, value, index) => {
+        let template = this.getUrl(value);
+        template.then((res) => {
+          this.templates[key] = res;
+        })
+      });
+    },
+
+    // GET data via AJAX or pass along data object
+    //
+    getData: function(){
+      if (!!this.options.data) {
+        this.setData(this.options.data);
+      } else {
+        let data = this.getUrl(this.options.dataUrl);
+        data.then((res) => {
+          this.setData(res)
+        });
+      }
+    },
+
+    // shortcut for AJAX get
+    getUrl: function(url) {
+      return $.get(url, (data) => {
+        return data;
+      });
     },
 
-    // method to model our raw data payload
+    // TODO: make this more universal
+    //
+    // method to cast our data payload into module's context
     // @param value [object] API payload returned via AJAX
     //
-    configureData: function(value){
-      // TODO: establish how we want to configure the data
+    setData: function(value){
       this.data = {
         next: value.next,
         previous: value.previous,
-        glossary: value.results[0].analysis_type.glossary
+        analysis_type: value.results[0].analysis_type
       }
     },
 
-    // hooks into Annotator lifecycle events
+    // returns a promise that isn't resolved until our survey is submitted
     //
-    // TODOS:
-    //  • how to create our own UI
-    //  • how best to store our answers to interact with other plugins
-    //  • how to pause the annotator lifecycle?
-    beforeAnnotationCreated: function(annotation){
-      return this.promiseHelper(annotation)
-        .then((value) => {
-          this.setUpCustomEditorPanel(value);
-        });
+    createSurvey: function(annotation) {
+      return new Promise((resolve, reject) => {
+        this.data.analysis_type.topics.forEach(this.getTopics.bind(this));
+        let data = {
+          thisTopics: this.survey.topics.join('')
+        }
+
+        let formTemplate = Handlebars.compile(this.templates.form)
+
+        $('body').append(formTemplate(data))
+
+        $('.submit').on('click', (e) => {
+          console.log($('form').serialize())
+          resolve(annotation)
+          return false
```

Comment:
to keep the button from doing its default action and reloading the page. same as `e.preventDefault()`